File: 3718_smallest_missing_multiple_of_k/solution.go
// LeetCode 3718 - Smallest Missing Multiple of K
// https://leetcode.com/problems/smallest-missing-multiple-of-k/

func missingMultiple(nums []int, k int) int {
	s := map[int]bool{}
	for _, x := range nums {
		s[x] = true
	}
	for i := 1; ; i++ {
		if x := k * i; !s[x] {
			return x
		}
	}
}

File: 3718_smallest_missing_multiple_of_k/solution.ts
// LeetCode 3718 - Smallest Missing Multiple of K
// https://leetcode.com/problems/smallest-missing-multiple-of-k/

export function missingMultiple(nums: any, k: any): any {
    const s = new Set(nums);
    for (let i = 1; ; i++) {
        const x = k * i;
        if (!s.has(x)) return x;
    }
}
